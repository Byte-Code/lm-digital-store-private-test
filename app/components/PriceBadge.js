import React, { Component } from 'react';
import ImmutablePropTypes from 'react-immutable-proptypes';
import styled from 'styled-components';

import StoreStockBadge from './StoreStockBadge';

const Wrapper = styled.div`
  width: 255px;
  display: flex;
  flex-direction: column;
  padding: 22px 10px 10px;
  background-color: #f7f7f7;
`;

const Price = styled.p`
  font-size: 36px;
  text-align: right;
  margin-bottom: 3px;
`;

const Quantity = styled.p`
  font-size: 14px;
  font-family: LeroyMerlinSans Light;
  text-align: right;
`;

const Divider = styled.div`
  border: 1px dashed #333333;
  width: 100%;
  margin: 15px 0;
`;

const Button = styled.div`
  width: '100%';
  text-transform: uppercase;
  background: ${props => props.bgColor};
  height: 60px;
  display: flex;
  justify-content: center;
  align-items: center;
  text-align: center;
  color: #fff;
  box-shadow:  0 0 8px 0 rgba(51, 51, 51, 0.1);
  cursor: pointer;
`;

const StoreStockWrapper = styled.div`
  margin-bottom: 20px;
`;

export default class PriceBadge extends Component {
  static propTypes = {
    pricingInfo: ImmutablePropTypes.map.isRequired,
    currentStoreStock: ImmutablePropTypes.map.isRequired
  }

  render() {
    const { pricingInfo, currentStoreStock } = this.props;
    const sellingPrice = pricingInfo.get('gross').toFixed(2);

    return (
      <Wrapper>
        <Price>{sellingPrice} &#8364;</Price>
        <Quantity>1 pz / pz</Quantity>
        <Divider />
        <StoreStockWrapper>
          <StoreStockBadge currentStoreStock={currentStoreStock} />
        </StoreStockWrapper>
        <Button bgColor="#67cb33">verifica disponibilit√† in negozi vicini</Button>
        <Divider />
        <Button bgColor="#339900">acquista online</Button>
      </Wrapper>
    );
  }

}
