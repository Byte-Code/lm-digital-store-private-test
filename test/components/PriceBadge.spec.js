import React from 'react';
import { shallow } from 'enzyme';
import { fromJS } from 'immutable';

import PriceBadge, { Discount } from '../../app/components/PriceBadge';

describe('PriceBadge', () => {
  it('should only render the price if no discount is active', () => {
    const price = fromJS({
      gross: 10
    });
    const pricingInfo = fromJS({
      sellingCapacity: 'unit',
      sellingUnit: 'unit'
    });
    const result = shallow(
      <PriceBadge
        price={price}
        pricingInfo={pricingInfo}
      />
    );
    const discount = result.find(Discount);
    expect(result).toMatchSnapshot();
    expect(discount).toHaveLength(0);
  });

  it('should also render the discount when it is active', () => {
    const price = fromJS({
      list: 15,
      gross: 10,
      discount: 5
    });
    const pricingInfo = fromJS({
      sellingCapacity: 'unit',
      sellingUnit: 'unit'
    });
    const result = shallow(
      <PriceBadge
        price={price}
        pricingInfo={pricingInfo}
      />
    );
    const discount = result.find(Discount);
    expect(result).toMatchSnapshot();
    expect(discount).toHaveLength(1);
  });
});
