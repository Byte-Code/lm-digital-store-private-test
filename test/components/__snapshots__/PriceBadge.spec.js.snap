// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PriceBadge should also render the discount when it is active 1`] = `
<div>
  <styled.div>
    <span>
      -
      5
       %
    </span>
    <span>
      15,00
      €
    </span>
  </styled.div>
  <styled.p
    isDiscounted={true}
  >
    10,00
     €
  </styled.p>
  <styled.p>
    unit unit / unit
  </styled.p>
</div>
`;

exports[`PriceBadge should only render the price if no discount is active 1`] = `
<div>
  <styled.p>
    10,00
     €
  </styled.p>
  <styled.p>
    unit unit / unit
  </styled.p>
</div>
`;
